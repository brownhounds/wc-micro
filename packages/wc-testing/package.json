{
    "name": "@brownhounds/wc-testing",
    "version": "0.1.0",
    "license": "MIT",
    "author": "brownhounds",
    "main": "dist/index.js",
    "types": "dist/index.d.ts",
    "files": [
        "./dist/**/*"
    ],
    "exports": {
        ".": "./dist/index.js",
        "./snapshots": "./dist/snapshots.js"
    },
    "scripts": {
        "clean": "rm -rf ./dist",
        "emit-declarations": "tsc --emitDeclarationOnly --declaration --outDir ./dist",
        "build-lib": "vite build",
        "build": "npm run clean && npm run build-lib && npm run emit-declarations",
        "lint": "tsc --noEmit && eslint --ext .ts ."
    },
    "devDependencies": {
        "@brownhounds/eslint-config": "0.1.0",
        "@brownhounds/ts-config": "0.1.0",
        "@brownhounds/uhtml": "*",
        "typescript": "5.3.3",
        "vite": "5.0.12",
        "vitest": "1.4.0",
        "rehype-format": "4.0.1",
        "rehype-parse": "8.0.4",
        "rehype-stringify": "9.0.3",
        "unified": "10.1.2"
    }
}
