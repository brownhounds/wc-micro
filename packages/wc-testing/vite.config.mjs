import { defineConfig } from 'vite';
import { resolve } from 'path';

export default defineConfig({
    build: {
        lib: {
            entry: [resolve('index.ts'), resolve('snapshots.ts')],
            name: 'wc-testing',
            fileName: (_, entryName) => `${entryName}.js`,
            formats: ['cjs'],
        },
    },
});
