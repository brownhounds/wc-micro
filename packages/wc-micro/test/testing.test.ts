import { expect, test, describe } from 'vitest';
import { Present } from '@brownhounds/wc-testing';
import { HTML } from '@brownhounds/wc-testing/snapshots';
import { html } from '@brownhounds/uhtml';

import './fixtures/MainApp';
import type { MainApp } from './fixtures/MainApp';

describe('test', () => {
    test('true', async () => {
        const presenter = await new Present().screen(html`<main-app />`);
        console.log(presenter.root<MainApp>().tagName);
        const t = HTML(presenter.root());
        expect(t).toMatchSnapshot();
    });
});
