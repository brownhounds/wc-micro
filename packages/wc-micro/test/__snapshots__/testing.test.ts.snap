// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`test > true 1`] = `
"
<main-app><!-- #shadowroot -->
  <h1>Hello There from MainApp!! State Value</h1>
  <test-ono><!-- #shadowroot -->
    <div>
      <div>kjfhksfdh</div>
      <h1>Nested Component!! State Value</h1>
    </div>
  </test-ono>
</main-app>
"
`;
